node/pkg/nodeinit/calico-init_linux.go

```diff
@@ -65,11 +74,52 @@ func ensureBPFFilesystem() error {
 	}
 
 	// If we get here, the BPF filesystem is not mounted.  Try to mount it.
-	logrus.Info("BPF filesystem is not mounted.  Trying to mount it...")
-	err = syscall.Mount(bpfFSMountPoint, bpfFSMountPoint, "bpf", 0, "")
+	logrus.Info("BPF filesystem is not mounted. Trying to mount it...")
+	err = syscall.Mount(bpf.DefaultBPFfsPath, bpf.DefaultBPFfsPath, "bpf", 0, "")
 	if err != nil {
 		return fmt.Errorf("failed to mount BPF filesystem: %w", err)
 	}
 	logrus.Info("Mounted BPF filesystem.")
 	return nil
 }
+
+const cgroupRootPath = "/initproc"
+
+// ensureCgroupV2Filesystem() enters the cgroup and mount namespace of the process
+// with PID 1 running on a host to allow felix running in calico-node access the root of cgroup namespace.
+// This is needed by felix to attach CTLB programs and implement k8s services correctly.
+func ensureCgroupV2Filesystem() error {
+	// Check if the Cgroup2 filesystem is mounted at the expected location.
+	logrus.Info("Checking if Cgroup2 filesystem is mounted.")
+	mountInfoFile := path.Join(cgroupRootPath, "mountinfo")
+	mounts, err := os.Open(mountInfoFile)
+	if err != nil {
+		return fmt.Errorf("failed to open %s: %w", mountInfoFile, err)
+	}
+	scanner := bufio.NewScanner(mounts)
+	for scanner.Scan() {
+		line := scanner.Text()
+		mountPoint := strings.Split(line, " ")[4] // 4 is the index to mount points in mountinfo files
+
+		if mountPoint == bpf.CgroupV2Path && strings.Contains(line, "cgroup2") {
```

Comment:
> `strings.Contains(line, "cgroup2")`

Please check the right "column" for this value.  We've been caught out before by doing `if strings.Contains(line, "bpf")` and then finding our that `bpf` appeared in an unexpected part of the string.

Reply:
fixed it. However, there is an assumption (also in mounting bpf fs function), that a space character is delimiter. I used the same assumption here. But I wonder if this is a correct assumption. WDYT?